Fix Phantoms

BONNe opened the pull request:
It looks like in Bukkit, Phantoms is not a Monster.

BONNe committed:
Fix issue when Phantoms did not spawn in BentoBox worlds.

It looks like in Bukkit, Phantoms is not a Monster.

BONNe committed:
Improve living mob processing.

Introduce 2 new methods in the Util class:
- isAnimalEntity: returns if a given entity must be protected by ANIMAL_FLAG
- isMonsterEntity: returns if a given entity must be protected by MONSTER_FLAG

This will fix the issue with unprotected mobs what are not implementing Monsters and Animals interfaces. These changes remove Pufferfish and Squid from monster protection flag.

All methods which used Entity instanceOf Monster are changed with Util.isMonsterEntity(Entity).
All methods which used Entity instanceOf Animal are changed with Util.isAnimalEntity(Entity).

Poslovitch reviewed (commented):
Let's postpone this PR as long as we do not have a clearly defined "policy" with all of those monster/animals things.

BONNe committed:
Change method names:

Util#isAnimalEntity() is changed to Util#isPassiveEntity()
Util#isMonsterEntity() is changed to Util#isHostileEntity()

Add Pufferfish to Hostile Entity processing.

BONNe committed:
Move pufferfish spawn handling from MONSTER_SPAWN flag to ANIMAL_SPAWN flag.

Poslovitch committed:
Improved javadoc
